src/Spectre.libGaussianMixtureModelling/GaussianDistribution.h

```diff
@@ -28,8 +28,47 @@ namespace spectre::unsupervised::gmm
 /// Computes value of Guassian Function, also known as Normal distribution
 /// based on given mean, standard deviation for a single observation.
 /// </summary>
+/// <param name="x">Argument to compute value for.</param>
+/// <param name="mean">Mean of the distribution.</param>
+/// <param name="std">Standard deviation of the distribution.</param>
 inline DataType Gaussian(DataType x, DataType mean, DataType std)
 {
-    return (1.0 / sqrt(2.0 * M_PI * std * std)) * exp(-pow(x - mean, 2) / (2.0 * std * std));
+    static const DataType inverseOfSqrtTwoPi = 1.0 / sqrt(2.0 * M_PI);
+    DataType twoExpSqrt = (x - mean) / std;
+    return inverseOfSqrtTwoPi * std::exp(-0.5 * twoExpSqrt * twoExpSqrt) / std;
 }
+
+/// <summary>
+/// Serves as a representation of Normal Distribution with fixed parameters.
+/// </summary>
+class GaussianDistribution
+{
+static constexpr DataType invSqrtOfTwoPi = 0.3989422804014327; // 1/sqrt(2pi)
+public:
+    /// <summary>
+    /// Constructs a distrubution based on given mean and standard deviation.
+    /// </summary>
+    /// <param name="mean">Mean of the distribution.</param>
+    /// <param name="std">Standard deviation of the distribution.</param>
+    GaussianDistribution(DataType mean, DataType std) : m_Mean(mean),
```

Comment:
We will need information on `std` and _weight_ in final model - this is used for components merging, etc. How do you preserve this? Do you return this at some point other way?

Reply:
These are saved in Gaussian Components.